data/tutorials/lg_00_data_types_and_matching.md

```diff
@@ -299,6 +299,76 @@ val insert : 'a * 'b -> ('a * 'b) tree -> ('a * 'b) tree = <fun>
 Similar functions can be written to look up values in a dictionary, to convert
 a list of pairs to or from a tree dictionary and so on.
 
+## Example: options
+
+The OCaml language has been supporting the [option
+type](https://en.wikipedia.org/wiki/Option_type) since its inception.
```

Comment:
The Wikipedia article reads:

> In [programming languages](https://en.wikipedia.org/wiki/Programming_language) (especially [functional programming](https://en.wikipedia.org/wiki/Functional_programming) languages) and [type theory](https://en.wikipedia.org/wiki/Type_theory), an option type or maybe type is a [polymorphic type](https://en.wikipedia.org/wiki/Parametric_polymorphism) that represents encapsulation of an optional value; e.g., it is used as the return type of functions which may or may not return a meaningful value when they are applied. It consists of a constructor which either is empty (often named None or Nothing), or which encapsulates the original data type A (often written Just A or Some A).

I was expecting something much more complicated, but still, I'm not sure it's the most beginner-friendly explanation we could give. E.g. this assumes that the reader knows what encapsulation means (and coming from OOP, what one understands by "encapsulation" can be something completely different!). Same for polymorphic types. I'd recommend writing a friendlier introduction of the option type in the doc.